testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestReadUniontype.java

```diff
@@ -139,6 +137,127 @@ public void testReadUniontype(String storageFormat)
+        if (getHiveVersionMajor() != 1 || getHiveVersionMinor() != 2) {
+            throw new SkipException("This test can only be run with Hive 1.2 (default config)");
+        }
+
+        onHive().executeQuery(format(
+                "CREATE TABLE %s ("
+                        + "c0 INT,"
+                        + "c1 UNIONTYPE<"
+                        + "     STRUCT<a:STRING, b:STRING>, "
+                        + "     STRUCT<c:STRING>>) "
+                        + "PARTITIONED BY (c2 INT) "
+                        + "STORED AS %s",
+                TABLE_NAME_SCHEMA_EVOLUTION,
+                storageFormat));
+
+        if (storageFormat.equals("AVRO")) {
+            /**
+             * The following insertion fails on avro.
+             *
+             * hive (default)> INSERT INTO TABLE  u_username.test_ut_avro partition (c2 = 5)
+             *               > SELECT 1, create_union(1, named_struct('a', 'ignore', 'b', 'ignore'), named_struct('c', 'c1'));
+             *
+             * Error: java.lang.RuntimeException: org.apache.hadoop.hive.ql.metadata.HiveException: Hive Runtime Error while processing writable (null)
+             * at org.apache.hadoop.hive.ql.exec.mr.ExecMapper.map(ExecMapper.java:179)
+             *  at org.apache.hadoop.mapred.MapRunner.run(MapRunner.java:54)
+             *  at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:459)
+             *  at org.apache.hadoop.mapred.MapTask.run(MapTask.java:343)
+             *  at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:177)
+             *  at java.security.AccessController.doPrivileged(Native Method)
+             *  at javax.security.auth.Subject.doAs(Subject.java:422)
+             *  at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1893)
+             *  at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:171)
+             * Caused by: org.apache.hadoop.hive.ql.metadata.HiveException: Hive Runtime Error while processing writable (null)
+             *  at org.apache.hadoop.hive.ql.exec.MapOperator.process(MapOperator.java:505)
+             *  at org.apache.hadoop.hive.ql.exec.mr.ExecMapper.map(ExecMapper.java:170)
+             *  ... 8 more
+             * Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
+             *  at org.apache.avro.generic.GenericData$Record.get(GenericData.java:135)
+             *  at org.apache.avro.generic.GenericData.getField(GenericData.java:580)
+             *  at org.apache.avro.generic.GenericData.validate(GenericData.java:373)
+             *  at org.apache.avro.generic.GenericData.validate(GenericData.java:395)
+             *  at org.apache.avro.generic.GenericData.validate(GenericData.java:373)
+             *  at org.apache.hadoop.hive.serde2.avro.AvroSerializer.serialize(AvroSerializer.java:96)
+             *
+             * So we try coercion logic on the first struct field only.
```

Comment:
can you mention `first struct field inside the union (i.e. only for <a,b> struct)` for clarity?